build.gradle

```diff
@@ -312,6 +312,7 @@ project(':iceberg-aws') {
     implementation project(':iceberg-core')
 
     compileOnly 'software.amazon.awssdk:url-connection-client'
+    compileOnly 'software.amazon.awssdk:apache-client'
```

Comment:
let's focus this PR around using URL connection client correctly. 

After this one we can introduce the use of Apache client. The reason is that we should keep the current behavior since users bring a `url-connection-client` in their environment will see ClassNotFoundError if we directly switch to `apache-client`. I think we should offer a catalog config in `AwsProperties`, something like `client.http.type`, where it could be `url` or `apache`. Does that work for you?

Reply:
True. Sounds good.